internal/util/helpers_test.go

```diff
@@ -4,18 +4,758 @@
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			expectedError: "error unmarshaling credentials: EOF",
+		},
+		{
+			name:          "Error unmarshaling credentials with local server",
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			responseBody:  `invalid_json`,
+			expectedError: "error unmarshaling credentials: invalid character 'i' looking for beginning of value",
+		},
+		{
+			name:          "Error refreshing credentials with local server",
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			responseBody:  `{"error":"invalid_grant","error_description":"Invalid refresh token"}`,
+			expectedError: "error refreshing credentials: invalid_grant: Invalid refresh token",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				fmt.Fprintln(w, tt.responseBody)
+			}))
+			defer server.Close()
+
+			parsedURL, _ := url.Parse(server.URL)
+			tt.issuerUrl = parsedURL.String()
+
+			result, err := util.RefreshCredentials(tt.refreshToken, tt.issuerUrl, tt.clientId)
+			if tt.expectedError != "" {
+				if err == nil || err.Error() != tt.expectedError {
+					t.Errorf("expected error %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+		})
+	}
+}
+
+// TestLoadCredentials tests the LoadCredentials function
+func TestLoadCredentials(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name           string
+		fileContent    string
+		expectedError  string
+		expectedResult util.OpenIdCredentials
+	}{
+		{
+			name:        "Successful load",
+			fileContent: `{"access_token":"access_token","refresh_token":"refresh_token","expiry":"2024-10-05T17:46:27+10:00"}`,
+			expectedResult: util.OpenIdCredentials{
+				AccessToken:          "access_token",
+				RefreshToken:         "refresh_token",
+				AccessTokenExpiresAt: time.Date(2024, 10, 5, 17, 46, 27, 0, time.FixedZone("AEST", 10*60*60)),
+			},
+		},
+		{
+			name:          "Error unmarshaling credentials",
+			fileContent:   `invalid_json`,
+			expectedError: "error unmarshaling credentials: invalid character 'i'",
+		},
+		{
+			name:          "Error reading credentials file",
+			fileContent:   "",
+			expectedError: "error reading credentials file",
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tempDir := t.TempDir()
+			defer os.RemoveAll(tempDir)
```

Comment:
You shouldn't need this if you use [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir):

> The directory is automatically removed when the test and all its subtests complete.

Reply:
Removed